### PR TITLE
Fix: buffer messages from external resource

### PR DESCRIPTION
**Pull request by nicomoya123**
- When setting the to messageEventName: 'messageBuffer', the channel is received as buffer. I have added a check to set it as such.
- The buffer is also serialized as json when so I have checked if it as a buffer and return it as it is.
תודה !

### CODE REVIEW

File: src/redis-pubsub.ts [hunk 1]
```diff
@@ -169,17 +175,23 @@ export class RedisPubSub implements PubSubEngine {
...
     // Don't work for nothing..
     if (!subscribers?.size) return;
 
     let parsedMessage;
     try {
-      parsedMessage = this.deserializer
-        ? this.deserializer(message, { pattern, channel })
-        : JSON.parse(message, this.reviver);
+      if(this.deserializer){
+        parsedMessage = this.deserializer(Buffer.from(message), { pattern, channel })
```
Comment (ujwal-setlur) [inline src/redis-pubsub.ts:189]: The serializer/deserializer may not be dealing with buffers at all. I think the original code is correct. I think the only change needed is the check above for whether channel is an object.

Comment (ujwal-setlur) [reply src/redis-pubsub.ts:189]: Yes, there are some serialization/desrialization libraries that don't necessarily output buffers or take in buffers. They could be encoding/decoding using strings (inefficient, I know). But then, I guess they would not be setting the eventName to messageBuffer in the config. So this is likely ok.

**Comment by nicomoya123** [comment]
@davidyaha can this be merged ?

**Comment by davidyaha** [comment]
Thanks @nicomoya123 and sorry for the delay

**Pull request merged**